.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run.
+on:
```

Comment:
we usually have 
```
on: [push, pull_request]
```
so its not limited to master branch only, but `workflow_dispatch` seems like a nice thing also

Reply:
makes sense, tweaked it